### ISSUE TITLE
Ensure msbuild from vs usage is validated in arcade or arcade-validation

### ISSUE THREAD

**Issue by riarenas** [dotnet/arcade#7567]
- [ ] This issue is blocking 
- [x] This issue is causing unreasonable pain

We had a bug that went through the entire validation and promotion process without being caught because it only surfaces when using the desktop msbuild from vs (not the xcopyable one)

See: https://github.com/dotnet/arcade/pull/7558 and https://github.com/dotnet/arcade/pull/7563 for context.

We should make sure we have some sort of validation (ideally as early as possible, during the arcade PR build) for this failure mode across branches.

**Comment by riarenas** [comment]
I would like this done as part of FR. These validation holes end up being really noisy when they manage to get to consumer repos. Not only do they end up blocking a dev's entire day while they follow the fix around and make sure it gets to repos, but it erodes the confidence in arcade updates. @ilyas1974 feel free to take it out if you think this is out of scope.

**Comment by ilyas1974** [comment]
I think we can handle this in FR. As this does not have direct customer impact at this time, it will be something we address as time allows.

**Comment by adiaaida** [comment]
Unfortunately, just passing in -msbuildEngine vs on an arcade build as an extra leg did not work. It complains that there is no vs object on the tools object in the global.json. So I grabbed a thing from aspnetcore (who does have the vs object) and tried adding that to arcade's global.json, which caused all of the windows builds to break, saying that it can't find the sdk.

```
It was not possible to find any installed .NET SDKs.
```

Going to try a few more things. Also curious how arcade validation doesn't already catch this, given that we do official builds on the important repos before promoting.

**Comment by riarenas** [comment]
>Also curious how arcade validation doesn't already catch this, given that we do official builds on the important repos before promoting.

We do not build any other repos anymore as part of promotion.

**Comment by adiaaida** [comment]
Oh.

**Comment by adiaaida** [comment]
Throwing this back to triage

**Comment by chcosta** [comment]
> Unfortunately, just passing in -msbuildEngine vs on an arcade build as an extra leg did not work. It complains that there is no vs object on the tools object in the global.json. So I grabbed a thing from aspnetcore (who does have the vs object) and tried adding that to arcade's global.json, which caused all of the windows builds to break, saying that it can't find the sdk.
> 
> ```
> It was not possible to find any installed .NET SDKs.
> ```
> 
> Going to try a few more things. Also curious how arcade validation doesn't already catch this, given that we do official builds on the important repos before promoting.

We've encountered this, or a variant of this, a couple of times where Arcade's support for framework msbuild is not on par with its support for core MSBuild and Arcade doesn't have support for mix /matching either scenario.  Arcade assumes you're all in on either scenario.  

Here's a summary from previous work to unblock a partner...

> The issue ended up being that in Jacques scenario, he was using the machine installed (VS) MSBuild, but was also restoring a repo local SDK.  

> Arcade has this behavior:
> 1)	It looks in global.json for “tools.vs”, and then it looks for “tools.dotnet”, to help decide the default MSBuild engine.  If both are defined, then “tools.vs” is used (on Windows).  Ie, you default to VS MSBuild.
> 2)	Arcade doesn’t support (well) the scenario where you want to mix / match global dependencies with local dependencies.  It’s either, use the machine wide tools, or install them all locally and use them there.  This mix / match is what led to the error “It was not possible to find any installed .NET SDKs.” Even though we could see the sdk being installed.

> The workarounds for Arcade behavior were this:
> 1)	For Arcade behavior (1) above.  The installer repo has many build legs, but only one of them is supposed to run on VS MSBuild, so we had to explicit set MSBuildEngine to “dotnet” for most of the build legs
> 2)	For Arcade behavior (2) above.  The way we worked around this was to define “MSBuildSdksPath” explicitly in the repo’s “eng\configure-toolset.ps1” file when running with VS MSBuild engine.

>> if ($msbuildEngine -eq 'vs')
>> {
>> $globalJson = Get-Content -Raw global.json | ConvertFrom-Json
>> $env:MSBuildSdksPath = [IO.Path]::Combine($RepoRoot, ".dotnet\sdk", $globalJson.tools.dotnet, "sdks")
>> }

> I don’t think that either of the fixes warrant a change in Arcade behavior at this time as neither scenario is particularly mainstream.  (2) is slightly more interesting as we don’t’ have a good way to use framework MSBuild with a specific SDK apart from requiring a machine wide install and that would be an overly high support cost.  

ie, because we haven't fixed any of this, we don't have a good way to support validating Framework MSBuild (VS), in the Arcade or Arcade-validation repos.

**Comment by chcosta** [comment]
Any objection to me making the administrator [dncengvars](https://idweb/IdentityManagement/aspx/common/GlobalSearchResult.aspx?searchtype=e0c132db-08d8-4258-8bce-561687a8a51e&content=dncengvars) instead.  That group was created because of https://github.com/dotnet/core-eng/issues/2937

**Comment by adiaaida** [comment]
Is this comment meant for your other issue?

**Comment by chcosta** [comment]
Yes!

**Comment by dougbu** [comment]
> which caused all of the windows builds to break

This is expected because the `$MSBuildEngine` will default to `vs` when the `tools.vs` exists in global.json. Could use `-msbuildEngine dotnet` in existing Windows builds.

**Comment by missymessa** [comment]
Still relevant (and painful) today. There may be an issue for better logging somewhere that might accomplish the same goal as this, however.

**Comment by chcosta** [comment]
This would be less painful if we invested in https://github.com/dotnet/arcade/issues/14430 because the only place we (Arcade) depend on the framework is in signtool (repos may have additional dependencies though).

https://github.com/dotnet/arcade/issues/14312, would also reduce the pain and would probably be sufficient.  If more is needed, then we could follow a pattern similar to https://github.com/dotnet/arcade/pull/14777 which would redirect the user to a wiki for certain error conditions and more info could be provided.

I don't think we're set up to invest in full framework testing for this at this time.
